src/ApplicationInsights.Kubernetes/K8sQueryClient.cs

```diff
@@ -91,12 +91,27 @@ public Task<IEnumerable<K8sDeployment>> GetDeploymentsAsync(CancellationToken ca
         #endregion
 
         #region Node
-        public Task<IEnumerable<K8sNode>> GetNodesAsync(CancellationToken cancellationToken)
+        public async Task<IEnumerable<K8sNode>> GetNodesAsync(bool ignoreForbiddenException, CancellationToken cancellationToken)
         {
             EnsureNotDisposed();
 
-            string url = Invariant($"api/v1/nodes");
-            return GetAllItemsAsync<K8sNode>(url, cancellationToken);
+            try
+            {
+                string url = Invariant($"api/v1/nodes");
+                return await GetAllItemsAsync<K8sNode>(url, cancellationToken).ConfigureAwait(false);
+            }
+            catch (UnauthorizedAccessException ex)
+            {
+                // Prefer ignoring the unauthorized access exception
+                if (ignoreForbiddenException)
+                {
+                    _logger.LogDebug(ex.Message);
+                    _logger.LogTrace(ex.ToString());
+                    return Enumerable.Empty<K8sNode>();
+                }
+                // Otherwise
+                throw;
```

Comment:
Nit: you can just say `else throw;`. No comment necessary ☺️

Reply:
I am sure you are okay with that, I just want to call out in case the return was oversighted. Thanks for the suggestion anyway.